### PR TITLE
Apply numpy docstyle to repo

### LINKED ISSUE #198: harmonize docstrings
We may actually want to bring this up to the wider HERA org, but at least within our own repo we should have some standard for docstring format. Personally I like the numpy format:
http://daouzli.com/blog/docstring.html#numpydoc

### PR DESCRIPTION
**Pull request by plaplant**
This PR applies the numpy docstring style to the main modules in the repo. It also adds a check to travis that will hopefully keep us from drifting too far away from it.

Anecdotally, pydocstyle is not quite as rigorous as pycodestyle is for PEP8. In the course of this PR, I found that I could have docstrings that did not adhere to the "typical" numpy template, and still passed the checks. I imagine this is a design choice, to allow for flexibility in the docstrings without being obnoxious. So conversely, it will likely take some due diligence on the human side of things to make sure we're sticking to the standard as much as possible.

@adampbeardsley @mkolopanis @nkern if you get a chance to look through these changes, I'd appreciate it. In many places I expanded on the existing docstrings, and I want to make sure I'm not accidentally lying to users.

Fixes #198.

### CODE REVIEW

File: hera_qm/ant_metrics.py [hunk 1]
```diff
@@ -123,21 +133,33 @@ def mean_Vij_metrics(data, pols, antpols, ants, bls,
...
+        indexing via data[i,j,pol].
+    pols
+        List of visibility polarizations (e.g. ['xx','xy','yx','yy']).
+    antpols
+        List of antenna polarizations (e.g. ['x', 'y']).
+    ants
+        List of all antenna indices.
+    bls
+        List of tuples of antenna pairs.
+    xants
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: https://github.com/HERA-Team/hera_qm/blob/7347c53d14d1cb6706e3d81c16102d494f90c6e3/hera_qm/ant_metrics.py#L138-L151

We are missing type definitions in the numpy style for these lines. (eg ` pols : list`) but at the same time it is quite obvious that it expects a list.

File: hera_qm/ant_metrics.py [hunk 2]
```diff
@@ -168,16 +190,22 @@ def mean_Vij_metrics(data, pols, antpols, ants, bls,
...
-              allowed values ['xx', 'yy', 'xy', 'yx']
-        reds: List of lists of tuples of antenna numbers that make up redundant baseline groups.
+    Parameters
+    ----------
+    data
+        Dictionary of visibility data. Keys are in the form (ant1, ant2, pol).
+    pols
+        List of polarizations to compute the median auto power. Allowed values
+        are ['xx', 'yy', 'xy', 'yx'].
+    reds
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: This functions again, doesn't explicitly state the type in the numpydoc style

File: hera_qm/ant_metrics.py [hunk 3]
```diff
@@ -192,27 +220,40 @@ def red_corr_metrics(data, pols, antpols, ants, reds, xants=[],
...
+    pols
+        List of visibility polarizations (e.g. ['xx','xy','yx','yy']).
+    antpols
+        List of antenna polarizations (e.g. ['x', 'y']).
+    ants
+        List of all antenna indices.
+    reds
+        List of lists of tuples of antenna numbers that make up redundant
+        baseline groups.
+    xants
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: This function has a mismatch of explicit types and not.

File: hera_qm/ant_metrics.py [hunk 4]
```diff
@@ -270,13 +311,20 @@ def red_corr_metrics(data, pols, antpols, ants, reds, xants=[],
...
-        xants: List of antennas ithat should be ignored.
-               format: (ant,antpol)
-    Retruns:
-        xantSet: List of all antenna-polarization combinations to exclude.
+    Parameters
+    ----------
+    antpols
+        List of Single antenna polarizations to add to excluded Antennas.
+        Should be one of ['x','y'], ['x'], or ['y'].
+    xants
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types.

File: hera_qm/ant_metrics.py [hunk 5]
```diff
@@ -289,23 +337,29 @@ def antpol_metric_sum_ratio(ants, antpols, crossMetrics, sameMetrics,
...
+        List of all antenna indices.
+    antpols
+        List of antenna polarizations (e.g. ['x', 'y']).
+    crossMetrics
+        Dict of a metrics computed with cross-polarizaed antennas. Keys are of
+        the form (ant, antpol).
+    sameMetrics
+        Dict of a metrics computed with non-cross-polarized antennas. Keys are of
+        the form (ant, antpol).
+    xants
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/ant_metrics.py [hunk 6]
```diff
@@ -324,30 +378,39 @@ def mean_Vij_cross_pol_metrics(data, pols, antpols, ants, bls, xants=[],
...
+        as data[i,j,pol].
+    pols
+        List of visibility polarizations (e.g. ['xx','xy','yx','yy']).
+    antpols
+        List of antenna polarizations (e.g. ['x', 'y']).
+    ants
+        List of all antenna indices.
+    bls
+        List of tuples of antenna pairs.
+    xants
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/ant_metrics.py [hunk 7]
```diff
@@ -377,27 +440,35 @@ def red_corr_cross_pol_metrics(data, pols, antpols, ants, reds, xants=[],
...
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/ant_metrics.py [hunk 8]
```diff
@@ -466,23 +550,55 @@ def load_antenna_metrics(filename):
...
-            reds: List of lists of tuples of antenna numbers that make up redundant baseline groups.
-            format: File type of data
-                    Supports: 'miriad', 'uvh5', 'uvfits', 'fhd', 'ms ' (see pyuvdata docs)
-                    Default: 'miriad'.
+        Parameters
+        ----------
+        dataFileList
+            List of data filenames of the four different visibility polarizations
+            for the same observation.
+        reds
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/ant_metrics.py [hunk 9]
```diff
@@ -466,23 +550,55 @@ def load_antenna_metrics(filename):
...
+        ants
+            List of antennas in HERAData object.
+        pols
+            List of polarizations in HERAData object.
+        bls
+            List of baselines in HERAData object.
+        dataFileList
+            List of data filenames of the four different visibility polarizations
+            for the same observation.
+        reds
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/ant_metrics.py [hunk 10]
```diff
@@ -560,25 +692,27 @@ def red_corr_metrics(self, pols=None, xants=[], rawMetric=False,
...
-        ratio of mean cross-pol visibilities to mean same-pol visibilities:
-        (Vxy+Vyx)/(Vxx+Vyy).
+        This method is a local wrapper for mean_Vij_cross_pol_metrics. It finds
+        which antennas are outliers based on the ratio of mean cross-pol visibilities
+        to mean same-pol visibilities:
+            (Vxy+Vyx)/(Vxx+Vyy).
+
+        Parameters
+        ----------
+        xants
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/ant_metrics.py [hunk 11]
```diff
@@ -588,26 +722,27 @@ def mean_Vij_cross_pol_metrics(self, xants=[], rawMetric=False):
...
-                                    and singlely-polarization flipped ones.
-                                    Very large values are probably cross-polarized.
+        This method is a local wrapper for red_corr_cross_pol_metrics. It finds
+        which antennas are part of visibilities that are significantly better
+        correlated with polarization-flipped visibilities in a redundant group.
+        It returns the modified z-score.
+
+        Parameters
+        ----------
+        xants
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/ant_metrics.py [hunk 12]
```diff
@@ -839,45 +981,55 @@ def ant_metrics_run(files, pols=['xx', 'yy', 'xy', 'yx'], crossCut=5.0,
...
+    type for a given JD, because the function will look for the other
+    polarizations in the same folder. If not all four polarizations are found,
+    a warning is generated, because the code assumes all four polarizations are
+    present.
+
+    Parameters
+    ----------
+    files
+        List of files to run ant metrics on. Can be any of the 4 polarizations.
+    pols
```
Comment (mkolopanis) [inline hera_qm/ant_metrics.py]: explicit types

File: hera_qm/firstcal_metrics.py [hunk 13]
```diff
@@ -831,14 +841,16 @@ def firstcal_metrics_run(files, args, history):
...
-    Return:
-        None
-
-    The function will take in a list of files and options. It will run the firstcal
+    This function will take in a list of files and options. It will run the firstcal
     metrics and produce a JSON file containing the relevant information.
+
+    Parameters
+    ----------
+    files
```
Comment (mkolopanis) [inline hera_qm/firstcal_metrics.py]: explicit types

File: hera_qm/metrics_io.py [hunk 14]
```diff
@@ -53,10 +53,15 @@ def _reds_list_to_dict(reds):
...
-        reds: List of list of tuples. e.g. list of list of antenna pairs for each baseline group
+    Parameters
+    ----------
+    reds
+        List of list of tuples. e.g. list of list of antenna pairs for each baseline group.
+
     Returns
-        reds: OrderedDict of baseline groups able to save to HDF5
+    -------
+    reds
```
Comment (mkolopanis) [inline hera_qm/metrics_io.py]: explicit types

File: hera_qm/metrics_io.py [hunk 15]
```diff
@@ -65,10 +70,15 @@ def _reds_list_to_dict(reds):
...
-        reds: OrderedDict of redundant baseline groups.
+    Parameters
+    ----------
+    reds
+        OrderedDict of redundant baseline groups.
+
     Returns
-        reds: List of lists of baseline groups.
+    -------
+    reds
```
Comment (mkolopanis) [inline hera_qm/metrics_io.py]: explicit types

File: hera_qm/metrics_io.py [hunk 16]
```diff
@@ -79,22 +89,36 @@ def _reds_dict_to_list(reds):
...
 
-    Arguments
-        h5file: H5py file object into which data is written, this is edited in place.
-        path: absolute path in HDF5 to store the current dictionary
-        in_dict: Dictionary to be recursively walked and stored in h5file
+    Parameters
+    ----------
+    h5file
+        An h5py file object into which data is written. This file is edited in place.
+    path
```
Comment (mkolopanis) [inline hera_qm/metrics_io.py]: explicit types

File: hera_qm/metrics_io.py [hunk 17]
```diff
@@ -260,13 +302,29 @@ def write_metric_file(filename, input_dict, overwrite=False):
...
     """Recursively read the hdf5 file and create sub-dictionaries.
 
-    Performs opposite function of _recursively_save_dict_to_group
+    This function performs the inverse operation of _recursively_save_dict_to_group.
+
+    Parameters
+    ----------
+    h5file
+        An h5py file object into which data is written. This file is edited in place.
+    path
```
Comment (mkolopanis) [inline hera_qm/metrics_io.py]: explicit types

File: hera_qm/metrics_io.py [hunk 18]
```diff
@@ -424,10 +494,15 @@ def _parse_value(in_val):
     Uses builtin types to make a last attempt to cast the value as an int, float, or complex.
     If all the types fail, returns a string.
 
-    Arguments
-        in_val: the unicode or string value to be parsed.
+    Parameters
+    ----------
+    in_val
```
Comment (mkolopanis) [inline hera_qm/metrics_io.py]: explicit types

File: hera_qm/metrics_io.py [hunk 19]
```diff
@@ -582,16 +692,22 @@ def _parse_dict_of_dicts(input_str, value_type=float):
...
+        The type to cast values in nested dictionaries to. Default is float.
 
-    Arguments
-        input_str: string to be processed
-        value_type: type to cast values in nested dictionaries to. Default
-        is float.
     Returns
-        output: dictionary of dictionary of dictionaries
+    -------
+    output
```
Comment (mkolopanis) [inline hera_qm/metrics_io.py]: explicit types

File: hera_qm/metrics_io.py [hunk 20]
```diff
@@ -614,12 +730,18 @@ def _parse_dict_of_dict_of_dicts(input_str, value_type=float):
...
+    ----------
+    input_str : str
+        The input string to be processed.
+    value_type
+        The type to cast values in nested dictionaries to. Default is float.
+
     Returns
-        output: dictionary of dictionary of dictionaries
+    -------
+    output
```
Comment (mkolopanis) [inline hera_qm/metrics_io.py]: explicit types

File: hera_qm/omnical_metrics.py [hunk 21]
```diff
@@ -155,16 +165,23 @@ def write_metrics(metrics, filename=None, filetype='json'):
...
+        multiple polarizations is passed in, only the
+        first polarization is used.
 
+    Returns
+    -------
+    fc_delays
+        The firstcal delays as saved in the file.
+    fc_gains
+        The firstcal delays in gains format.
+    fc_pol
```
Comment (mkolopanis) [inline hera_qm/omnical_metrics.py]: explicit types

File: hera_qm/omnical_metrics.py [hunk 22]
```diff
@@ -848,19 +963,26 @@ def plot_metrics(self, metrics):
...
 
-    Input:
-    ------
-    files : a list of *omni.calfits file to run metrics on
+    This function will produce a JSON file containing the series of metrics.
 
-    args : parsed arguments via argparse.ArgumentParser.parse_args
+    Parameters
+    ----------
+    files
```
Comment (mkolopanis) [inline hera_qm/omnical_metrics.py]: explicit types

File: hera_qm/utils.py [hunk 23]
```diff
@@ -32,361 +33,381 @@ def _str_to_bytes(s):
...
+
+    Parameters
+    ----------
+    method_name : {"ant_metrics", "firstcal_metrics", "omnical_metrics", "xrfi_run",
+                   "xrfi_apply", "xrfi_h1c_run", "delay_xrfi_h1c_idr2_1_run"}
+        The target wrapper desired.
+
+    Returns
+    -------
+    ap
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit type

File: hera_qm/utils.py [hunk 24]
```diff
@@ -32,361 +33,381 @@ def _str_to_bytes(s):
...
+                              'mean_Vij_cross_pol_metrics and '
+                              'red_corr_cross_pol_metrics are run. '
+                              'Default: True'))
+        ap.add_argument('--skip_cross_pols', action='store_false',
+                        dest='run_cross_pols',
+                        help=('Sets boolean flag to False. Flag determines if '
+                              'mean_Vij_cross_pol_metrics and '
+                              'red_corr_cross_pol_metrics are run. '
+                              'Default: True'))
+        ap.set_defaults(run_cross_pols=True)
```
Comment (mkolopanis) [inline hera_qm/utils.py]: YAY :tada: for no single letter variables

File: hera_qm/utils.py [hunk 25]
```diff
@@ -603,13 +638,20 @@ def metrics2mc(filename, ftype):
...
+    """Calculate the lst_array for a UVData or UVCal object.
+
+    Parameters
+    ----------
+    uv
+        A UVData or UVCal object.
+
+    Returns
+    -------
+    lst_array
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 26]
```diff
@@ -621,16 +663,30 @@ def lst_from_uv(uv):
...
+    ----------
+    a : array
+        Input array to process.
+    alg : str
+        Algorithm to use. Must be defined in this function with corresponding
+        subfunction below.
+    weights : array, optional
+        The weights to be used for the collapse operation (e.g. weighted mean).
+        NOTE: Some subfunctions do not use the weights. See corresponding doc strings.
+    axis
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 27]
```diff
@@ -648,16 +704,32 @@ def collapse(a, alg, weights=None, axis=None, returned=False):
...
+        The axis keyword to pass to np.sum.
+    returned : bool
+        If True, return sum of weights in addition to combined array. Default is False.
+
+    Returns
+    -------
+    o
+        The average of the input data.
+    wo
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 28]
```diff
@@ -675,24 +747,51 @@ def mean(a, weights=None, axis=None, returned=False):
...
+    axis
+        The axis keyword to pass to np.mean.
+    returned : bool
+        If True, return sum of weights in addition to combined array. Default is False.
+
+    Returns
+    -------
+    o
+        The average of the absolute value of the input data.
+    wo
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 29]
```diff
@@ -675,24 +747,51 @@ def mean(a, weights=None, axis=None, returned=False):
...
+    axis
+        The axis keyword to pass to np.mean.
+    returned : bool
+        If True, return sum of weights in addition to combined array. Default is False.
+
+    Returns
+    -------
+    o
+        The average in quadrature of the input data.
+    wo
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 30]
```diff
@@ -701,14 +800,28 @@ def quadmean(a, weights=None, axis=None, returned=False):
...
+        The axis or axes over which to OR.
+    returned : bool
+        If True, return dummy weights array. NOTE: the dummy weights will simply be
+        an array of ones. Default is False.
+
+    Returns
+    -------
+    o
+        The output of the OR operation on the input data.
+    wo
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 31]
```diff
@@ -721,14 +834,28 @@ def or_collapse(a, weights=None, axis=None, returned=False):
...
+        The axis or axes over which to AND.
+    returned : bool
+        If True, return dummy weights array. NOTE: the dummy weights will simply
+        be an array of ones. Default is False.
+
+    Returns
+    -------
+    o
+        The output of the AND operation on the input data.
+    wo
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 32]
```diff
@@ -741,13 +868,22 @@ def and_collapse(a, weights=None, axis=None, returned=False):
...
+    is (Ntimes, Nfreqs).
+
+    Parameters
+    ----------
+    waterfall
+        2D boolean array of shape (Ntimes, Nfreqs).
+
+    Returns
+    -------
+    wf
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/utils.py [hunk 33]
```diff
@@ -757,20 +893,31 @@ def and_rows_cols(waterfall):
...
+    into a single 2D waterfall of size (Ntimes, Nfreqs). To accomplish this, the
+    function averages over baselines and polarizations (in the case of visibility
+    data), or antennas and Jones parameters (in the case of calibration data).
+
+    Parameters
+    ----------
+    uv
+        A UVData or UVCal object which defines the times and frequencies, and
+        supplies the flag_array to convert (if flag_array not specified)
+    flag_array
```
Comment (mkolopanis) [inline hera_qm/utils.py]: explicit types

File: hera_qm/uvflag.py [hunk 34]
```diff
@@ -17,26 +18,79 @@
...
-            waterfall: Whether to immediately initialize as a waterfall object,
-                       with flag/metric axes: time, frequency, polarization. Default is False.
-            history: History string to attach to object.
-            label: string used for labeling the object (e.g. 'FM')
-        '''
+        """Initialize a UVFlag object.
+
+        Parameters
+        ----------
+        input
```
Comment (mkolopanis) [inline hera_qm/uvflag.py]: explicit types

File: hera_qm/uvflag.py [hunk 35]
```diff
@@ -426,26 +566,41 @@ def clear_unused_attributes(self):
...
-                metrics with this one.
-            method (str, optional): Method to combine metrics. Default is "quadmean".
-            inplace (bool, optional): Perform combination in place. Default is True.
-        Returns:
-            uvf (UVFlag): If inplace==False, return new UVFlag object with combined metrics.
+        """Combine metric arrays between different UVFlag objects together.
+
+        Parameters
+        ----------
+        others
```
Comment (mkolopanis) [inline hera_qm/uvflag.py]: explicit types

File: hera_qm/uvflag.py [hunk 36]
```diff
@@ -555,14 +715,31 @@ def collapse_pol(self, method='quadmean'):
...
-                       Otherwise, will require polarizations match.
-        '''
+        """Convert a UVFlag object of type "waterfall" to type "baseline".
+
+        This operation broadcasts the flag array to all baselines. This function
+        does NOT apply flags to uv.
+
+        Parameters
+        ----------
+        uv
```
Comment (mkolopanis) [inline hera_qm/uvflag.py]: explicit types

File: hera_qm/uvflag.py [hunk 37]
```diff
@@ -611,14 +788,31 @@ def to_baseline(self, uv, force_pol=False):
...
-                       Otherwise, will require polarizations match.
-        '''
+        """Convert a UVFlag object of type "waterfall" to type "antenna".
+
+        This operation broadcasts the flag array to all antennas. This function
+        does NOT apply flags to uv.
+
+        Parameters
+        ----------
+        uv
```
Comment (mkolopanis) [inline hera_qm/uvflag.py]: explicit types

File: hera_qm/uvflag.py [hunk 38]
```diff
@@ -711,37 +910,81 @@ def to_metric(self, convert_wgts=False):
...
+        Parameters
+        ----------
+        ant1 : int
+            Number of first antenna.
+        ant2 : int
+            Number of second antenna.
+
+        Returns
+        -------
+        ind
```
Comment (mkolopanis) [inline hera_qm/uvflag.py]: explicit types

File: hera_qm/uvflag.py [hunk 39]
```diff
@@ -711,37 +910,81 @@ def to_metric(self, convert_wgts=False):
...
         return uvutils.baseline_to_antnums(baseline, self.Nants_telescope)
 
     def get_baseline_nums(self):
-        """
-        Returns numpy array of unique baseline numbers in data.
+        """Return a numpy array of unique baseline numbers in data.
+
+        Returns
+        -------
+        baselines
```
Comment (mkolopanis) [inline hera_qm/uvflag.py]: explicit types

File: hera_qm/uvflag.py [hunk 40]
```diff
@@ -711,37 +910,81 @@ def to_metric(self, convert_wgts=False):
...
         return np.unique(self.baseline_array)
 
     def get_antpairs(self):
-        """
-        Returns list of unique antpair tuples (ant1, ant2) in data.
+        """Return a list of unique antpair tuples (ant1, ant2) in data.
+
+        Returns
+        -------
+        antnums
```
Comment (mkolopanis) [inline hera_qm/uvflag.py]: explicit types

File: hera_qm/xrfi.py [hunk 41]
```diff
@@ -23,14 +24,33 @@
...
-    Returns:
-        uvo: if inplace, applies flags to input uv. If not inplace,
-                uvo is a new UVFlag object with only xants flags.
+
+    Parameters
+    ----------
+    uv
+        Object containing data to be flagged. Should be a UVData, UVCal, or
+        UVFlag object.
+    xants
```
Comment (mkolopanis) [inline hera_qm/xrfi.py]: explicit types

File: hera_qm/xrfi.py [hunk 42]
```diff
@@ -531,21 +703,41 @@ def flag(uvf_m, nsig_p=6., nsig_f=None, nsig_t=None, avg_method='quadmean'):
...
-    '''
+    """Apply flags from UVFlag or list of UVFlag objects to UVData or UVCal.
+
+    Parameters
+    ----------
+    uvf
+        A UVFlag object, path to UVFlag file, or list of these. These must be in
+        'flag' mode, and either match the uv argument, or be a waterfall that can
+        be made to match it.
+    uv
```
Comment (mkolopanis) [inline hera_qm/xrfi.py]: explicit types

File: hera_qm/xrfi.py [hunk 43]
```diff
@@ -582,19 +774,36 @@ def flag_apply(uvf, uv, keep_existing=True, force_pol=False, history='',
...
-                  'gain', 'chisq', and 'tot_chisq' to use the gain_array,
-                  'quality_array', and 'total_quality_array', respectively.
-        **kwargs: Keyword arguments that are passed to algorithm.
-    Returns:
-        uvf: UVFlag object of mode 'metric' corresponding to the uv object.
+    """Make a UVFlag object of mode 'metric' from a UVData or UVCal object.
+
+    Parameters
+    ----------
+    uv
```
Comment (mkolopanis) [inline hera_qm/xrfi.py]: explicit types

File: hera_qm/xrfi.py [hunk 44]
```diff
@@ -648,28 +857,48 @@ def calculate_metric(uv, algorithm, cal_mode='gain', **kwargs):
...
-        uvf_f: UVFlag object of initial flags (initial flag + watershed)
-        uvf_wf: UVFlag object of waterfall type after thresholding in time/freq
-        uvf_w (if return_summary): UVFlag object with fraction of flags in uvf_f
+    """Run the xrfi excision pipeline we used for H1C.
+
+    This pipeline uses the detrending and watershed algorithms above.
+
+    Parameters
+    ----------
+    uv
```
Comment (mkolopanis) [inline hera_qm/xrfi.py]: explicit types

File: hera_qm/xrfi.py [hunk 45]
```diff
@@ -690,26 +919,41 @@ def xrfi_h1c_pipe(uv, Kt=8, Kf=8, sig_init=6., sig_adj=2., px_threshold=0.2,
...
-        uvf_m (UVFlag): UVFlag object with metric after collapsing to waterfall and to
-            single pol. Weights array is set to ones.
-        uvf_fws (UVFlag): UVFlag object with flags after watershed.
+    """Run the xrfi excision pipeline used for H1C IDR2.2.
+
+    This pipeline uses the detrending and watershed algorithms above.
+
+    Parameters
+    ----------
+    uv
```
Comment (mkolopanis) [inline hera_qm/xrfi.py]: explicit types

File: hera_qm/xrfi.py [hunk 46]
```diff
@@ -908,46 +1177,86 @@ def xrfi_h1c_run(indata, history, infile_format='miriad', extension='flags.h5',
...
+
+    Notes
+    -----
     This function will take in a UVData object or data file and optionally a cal file and
     model visibility file, and run an RFI-flagging algorithm to identify contaminated
     observations. Each set of flagging will be stored, as well as compressed versions.
+
+    Parameters
+    ----------
+    indata
```
Comment (mkolopanis) [inline hera_qm/xrfi.py]: explicit types

File: hera_qm/xrfi.py [hunk 47]
```diff
@@ -1059,29 +1368,52 @@ def xrfi_h1c_apply(filename, history, infile_format='miriad', xrfi_path='',
...
+        The path to save output to. Default is same directory as input file.
+    outfile_format : {"miriad", "uvfits", "uvh5"}
+        The file format for output files. Default is "miriad".
+    extension : str
+        The extension to be appended to input file name. Default is "R".
+    overwrite : bool
+        If True, overwrite the output file if it already exists. Default is False.
+    flag_file : str
+        The path to the npz file containing full flag array to insert into data file.
+    waterfalls
```
Comment (mkolopanis) [inline hera_qm/xrfi.py]: explicit types

File: hera_qm/utils.py (hunk 24 above)
Comment (adampbeardsley) [reply hera_qm/utils.py]: hey now, there are still plenty of single letter variables.

**Pull request merged**